### PR TITLE
PI-652 new staff details endpoint

### PR DESCRIPTION
**Pull request by stevomcallister**

### CODE REVIEW

File: projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/StaffService.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@Service
+class StaffService(
+    private val staffRepository: StaffRepository,
+) {
+    fun getOfficerView(code: String): OfficerView {
+        val staff = staffRepository.findByCode(code) ?: throw NotFoundException("Staff", "code", code)
+        return OfficerView(
+            code,
+            Name(staff.forename, staff.middleName, staff.surname),
+            staff.grade?.description,
```
Comment (marcus-bcl) [inline projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/StaffService.kt:19]: There are a couple of extension functions for these:
```suggestion
            staff.name(),
            staff.grade(),
```

File: projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/StaffService.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            null,
+            staffRepository.getSentencesDueCountByStaffId(staff.id, LocalDate.now().minusWeeks(4)),
```
Comment (marcus-bcl) [inline projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/StaffService.kt:21]: Should this be `dateTo = LocalDate.now().plusWeeks(4)`?

File: projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/provider/StaffRepository.kt [hunk 3]
```diff
@@ -18,4 +19,72 @@ interface StaffRepository : JpaRepository<Staff, Long> {
...
+       select 
+        count(1) 
+        from offender_manager om 
+                join event e on e.offender_id = om.offender_id
+                join disposal d on d.event_id = e.event_id
+                join custody c on d.disposal_id = c.disposal_id
+                join Institutional_report i on i.custody_id = c.custody_id
+                join r_standard_reference_list r on i.institution_report_type_id = r.standard_reference_list_id
+                where om.allocation_staff_id = :staffId
+                and om.end_date is null
```
Comment (marcus-bcl) [inline projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/provider/StaffRepository.kt:77]: Would it be better to check `om.active_flag = 1` here?

File: projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/model/OfficerView.kt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    val code: String,
+    val name: Name,
+    val grade: String?,
+    val email: String?,
+    val casesDueToEndInNext4Weeks: Long,
+    val releasesWithinNext4Weeks: Long,
+    val paroleReportsToCompleteInNext4Weeks: Long,
+)
+
+data class Team(
```
Comment (marcus-bcl) [inline projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/model/OfficerView.kt:13]: Is this needed?

**Review by marcus-bcl — CHANGES_REQUESTED** [review]
Looks good, think the logic is the wrong way round for "things due in the next four weeks" though.

Also probably worth adding an integration test?  The native queries should work with H2 too.

File: projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/model/OfficerView.kt (hunk 4 above)
Comment (stevomcallister) [reply projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/model/OfficerView.kt:13]: nope

File: projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/provider/StaffRepository.kt (hunk 3 above)
Comment (stevomcallister) [reply projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/provider/StaffRepository.kt:77]: ok

**Review by marcus-bcl — APPROVED** [review]
Looks good, and makes sense that the type is CUSTODY now, since a custody record is being added by the data loader.

**Pull request merged**